modules/core/src/test/java/org/apache/ignite/internal/processors/cache/transactions/TxWithKeyContentionSelfTest.java

```diff
@@ -216,120 +198,66 @@ public void testOptimisticRepeatableReadCheckContentionTxMetricNear() throws Exc
      * @param isolation Isolation level.
      * @throws Exception If failed.
      */
-    private void runKeyCollisionsMetric(TransactionConcurrency concurrency, TransactionIsolation isolation) throws Exception {
+    private void runKeyCollisionsMetric(TransactionConcurrency concurrency, TransactionIsolation isolation, boolean nearCashe)
+            throws Exception {
         Ignite ig = startGridsMultiThreaded(3);
 
         int contCnt = (int)U.staticField(IgniteTxManager.class, "COLLISIONS_QUEUE_THRESHOLD") * 5;
 
-        CountDownLatch txLatch = new CountDownLatch(contCnt);
-
         ig.cluster().state(ClusterState.ACTIVE);
 
         client = true;
 
         Ignite cl = startGrid();
 
-        IgniteTransactions cliTxMgr = cl.transactions();
-
-        IgniteCache<Integer, Integer> cache = ig.cache(DEFAULT_CACHE_NAME);
-
-        IgniteCache<Integer, Integer> cache0 = cl.cache(DEFAULT_CACHE_NAME);
+        IgniteCache<Integer, Integer> cache = ig.getOrCreateCache(getCacheConfiguration(nearCashe));
 
         final Integer keyId = primaryKey(cache);
 
-        CountDownLatch blockOnce = new CountDownLatch(1);
-
-        for (Ignite ig0 : G.allGrids()) {
-            if (ig0.configuration().isClientMode())
-                continue;
-
-            TestRecordingCommunicationSpi commSpi0 =
-                (TestRecordingCommunicationSpi)ig0.configuration().getCommunicationSpi();
-
-            commSpi0.blockMessages(new IgniteBiPredicate<ClusterNode, Message>() {
-                @Override public boolean apply(ClusterNode node, Message msg) {
-                    if (msg instanceof GridNearTxFinishResponse && blockOnce.getCount() > 0) {
-                        blockOnce.countDown();
-
-                        return true;
-                    }
-
-                    return false;
-                }
-            });
-        }
-
-        IgniteInternalFuture f = GridTestUtils.runAsync(() -> {
-            try (Transaction tx = cliTxMgr.txStart(concurrency, isolation)) {
-                cache0.put(keyId, 0);
-                tx.commit();
-            }
-        });
-
-        blockOnce.await();
+        IgniteTransactions transactions = cl.transactions();
 
-        GridCompoundFuture<?, ?> finishFut = new GridCompoundFuture<>();
+        AtomicBoolean doTest = new AtomicBoolean(true);
 
-        for (int i = 0; i < contCnt; ++i) {
-            IgniteInternalFuture f0 = GridTestUtils.runAsync(() -> {
-                try (Transaction tx = cliTxMgr.txStart(concurrency, isolation)) {
-                    cache0.put(keyId, 0);
+        IgniteInternalFuture f0 = GridTestUtils.runMultiThreadedAsync(() -> {
+            while (doTest.get()) {
+                try (Transaction tx = transactions.txStart(concurrency, isolation)) {
+                    cache.put(keyId, 0);
```

Comment:
`cache` is an instance of server node cache, `transactions` is an instance of client node transactions. This block is not working as you expected.